src/Libraries/CoreNodes/List.cs

```diff
@@ -525,6 +525,173 @@ public static IDictionary GroupByKey(IList list, IList keys)
             };
         }
 
+        /// <summary>
+        ///     This node clusters elements in a list based on the adjacency of their indices and the similarity of their values.
+        ///     It can handle list of numbers as well as list of strings.
+        ///     The boolean input allows the user to specify whether adjacency should be considered when grouping the elements.
+        ///     The tolerance input allows the user to specify the threshold value for grouping similar elements. In case of integers, the tolerance value is the difference between the two integers.
+        ///     And in case of strings, the tolerance value is the number of characters that can be different between the two strings.
+        /// </summary>
+        /// <param name="list">List of items to group as sublists based on adjacency and similarity</param>
+        /// <param name="tolerance">Threshold value for grouping similar items</param>
+        /// <param name="considerAdjacency">Boolean value to control if the node should consider adjacency or not.</param>
+        /// <returns name="groupedValues">list of sublists with items grouped by similar values</returns>
+        /// <returns name="groupedIndices">list of sublists containing the original indices of grouped values</returns>
+        /// <search>list;group;similar;adjacent;adjacency;groupbyadjacency;groupadjacentitems;groupsimilaritems;cluster;tolerance</search>
+        [MultiReturn(new[] { "groupedValues", "groupedIndices" })]
+        public static Dictionary<string, object> GroupBySimilarity(IList list, double tolerance = 0, bool considerAdjacency = true)
```

Comment:
I'm sorry, I didn't realize this function is in `List` class, which is marked as [`IsVisibleInLibrary(false)`](https://github.com/DynamoDS/Dynamo/blob/4106ba05b1dfa4fddba45991d57ea4b2d58bff71/src/Libraries/CoreNodes/List.cs#L17). This is a special case, in which case the class itself is hidden but all of its methods are not. So you were right at first; I think you need to explicitly add the attribute back - `IsVisibleInLibrary(true)`